### ISSUE TITLE
Option for `xsv cat row` to raise error if input files differ in field names

### ISSUE THREAD

**Issue by jeetsukumaran** [BurntSushi/xsv#5]
An error is raised if records are different lengths (differing number of columns/field) across input files, but is silent if the number of columns/fields are the same even if the names are different.

Could we have a flag that forces an error if the input files have columns of different names, even if the number is the same?

**Comment by BurntSushi** [comment]
That actually seems like pretty reasonable default behavior. Perhaps the flag should be for *disabling* that error check?

**Comment by jeetsukumaran** [comment]
You are right: the other way around makes more sense.

By the way, *great*, *great*, *great* toolkit. I use CSV files a *LOT*, and even started my toolkit to do some of these things, but was so relieved to find yours because it saved me a lot of work! THANK you !

**Comment by BurntSushi** [comment]
@jeetsukumaran Thanks. :-)

Out of curiosity, have you tried/heard of [`csvkit`](https://csvkit.readthedocs.org/en/0.9.0/)? If so, what did you find lacking about it? (For me it was mostly performance, but also some UI complaints.)

**Comment by jeetsukumaran** [comment]
Yes, I did. I discovered it after I started down my own (aborted) toolkit, as with yours, and I actually have it installed side-by-side with yours.

I like what csvkit does, but yes, it is a UI thing:

(1) I am all for "one program that does one thing well", but the fact that there were lots of little programs made things tricky. It was just difficult to always get a list of available csv* programs, and the figure out what it does. I so much prefer the more modern "`<program> <subcommand> ...`" (like `git` does) approach to `<program-subcommand>` approach.

(2) The other minor quibble is that many csvkit programs take a single file from the command like as an argument. So if I want to, e.g. check the headers of multiple files, I have to `for f in *.csv; do csv ... $f  ; done` instead of `xsv headers *.csv`. 

So far, the negative for me about xsv is the Rust dependency for building it (which is an unfair criticism, I know!). I have no problems having it setup on my personal machines, but I work with a number of clusters on which I have no administrative rights. Worse, some of these are *REALLY* old, with many out-of-date libraries, so building Rust there has proven problematical. I tried the pre-compiled binary you provide, but I get [this issue](https://github.com/BurntSushi/xsv/issues/3) on the old cluster. Looking forward to a fully static  build of `xsv`! 

As I said, I have both tools side-by-side, and I end up using `xsv` 90% of time because the UI is easier to work with.
